BlockchainSdk/Common/SS58.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  SS58.swift
+//  BlockchainSdk
+//
+//  Created by Dmitry Fedorov on 24.04.2024.
+//  Copyright © 2024 Tangem AG. All rights reserved.
+//
+
+import Foundation
+import Sodium
+
+/// Set of tools for working with ss58
+enum SS58 {
+    /// If data is bigger than 32 bytes we need to hash it, otherwise just return plain data
+    /// - Parameter data: public key data
+    /// - Returns: either plain or hashed data
+    static func accountData(from data: Data) -> Data {
+        guard data.count > Constants.publicKeySize else { return data }
+        return data.blake2hash(outputLength: Int(Constants.publicKeySize))
+    }
+    
+    /// The network type from provided address
+    /// see https://github.com/paritytech/ss58-registry/blob/main/ss58-registry.json `prefix` value
+    /// - Parameter address: Base58 encoded address, e.g. j4UiniaZ4GBrMQmjFPapb4fKMmXHxsPbVF4Lny1jHXSuWBKTS
+    /// - Returns: The network type
+    static func networkType(from address: String) throws -> UInt {
+        let decodedData = address.base58DecodedData
+        
+        guard decodedData.count >= Constants.maxPrefixSize else {
+            throw Error.invalidAddress
+        }
+        
+        let first = UInt(decodedData[0])
+        
+        guard first >= Constants.rangeLimit else {
+            return first
+        }
+        
+        let second = UInt(decodedData[1])
```

Comment:
и в этих местах думаю можно UInt8 сделать 

```suggestion
        let first = UInt8(decodedData[0])
        
        guard first >= Constants.rangeLimit else {
            return first
        }
        
        let second = UInt8(decodedData[1])
```

Reply:
возвращаемое значение может быть больше разрядности UInt8, ниже есть битовые операции, которые могут генерить как раз большие числа, в них придется все равно добавлять конвертацию, так что не вижу смысла.

Reply:
окей)